src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/InteropServices/CSharpDisableRuntimeMarshalling.Fixer.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public sealed class CSharpDisableRuntimeMarshallingFixer : CodeFixProvider
+    {
+        private class CustomFixAllProvider : DocumentBasedFixAllProvider
+        {
+            public static readonly CustomFixAllProvider Instance = new();
+
+            protected override string CodeActionTitle => MicrosoftNetCoreAnalyzersResources.UseDisabledMarshallingEquivalentCodeFix;
+
+            protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document, ImmutableArray<Diagnostic> diagnostics)
+            {
+                if (document.Project.CompilationOptions is CSharpCompilationOptions { AllowUnsafe: false })
+                {
+                    // We can't code fix if unsafe code isn't allowed.
+                    return await document.GetSyntaxRootAsync(fixAllContext.CancellationToken);
+                }
```

Comment:
Is this redundant? This was already checked in `RegisterCodeFixesAsync`

Reply:
I tried turning this into an assert and it tripped on a few tests, so I guess it's not redundant.